frontend/src/pages/user/admin/allReports/[adminReportID].tsx

```diff
@@ -2,88 +2,109 @@ import '@style/global.css';
 
 import { Col, Row } from 'react-bootstrap';
 import { useParams } from 'react-router-dom';
+import { useQuery } from '@tanstack/react-query';
+import { useNavigate } from 'react-router-dom';
 
-import SellerItem from '@components/SellerItem';
+import SellerItem, { SellerItemProps } from '@components/SellerItem';
 import NotFound from '@components/NotFound';
+import { CheckFetchStatus, RouteOnNotOK } from '@lib/Status';
 
-import adminReportData from '@pages/user/admin/adminReportData.json';
-
-interface Sellers {
-  id: number;
-  name: string;
-  imgUrl: string;
-  amount: number;
+interface SellersProps extends SellerItemProps {
+  total_sales: number;
 }
 
-interface Props {
-  year: number;
+interface AdminReportProps {
   month: number;
-  sellers: Sellers[];
-  totalAmount: number;
-  id: number;
+  sellers: SellersProps[];
+  total: number;
+  year: number;
 }
 
+const reportPageStyle = {
+  borderRadius: '50px 50px 0px 0px',
+  border: '1px solid var(--button_border)',
+  background: 'var(--bg)',
+  boxShadow: '0px 4px 30px 2px var(--title)',
+  padding: ' 7% 12% 10% 12%',
+};
+
 const AdminReportEach = () => {
-  const reportPageStyle = {
-    borderRadius: '50px 50px 0px 0px',
-    border: '1px solid var(--button_border)',
-    background: 'var(--bg)',
-    boxShadow: '0px 4px 30px 2px var(--title)',
-    padding: ' 7% 12% 10% 12%',
-  };
+  const navigate = useNavigate();
 
   const { year, month } = useParams();
+  const yearString = year ?? '';
+  const monthString = month ?? '';
 
-  const data: Props | undefined = adminReportData.find(
-    (item) => item.year.toString() === year && item.month.toString() === month,
-  );
+  const rfc3339Date = new Date(
+    Date.UTC(parseInt(yearString), parseInt(monthString) - 1, 1),
```

Comment:
all years and months are from selection, if someone type `http://localhost:8080/user/seller/reports/2dew/1d`, it will simply gives you `404 page not found` on webpage, i think that's enough

Reply:
> does this need error handling here

no, since the year and month are from select and the options are generated by us ( can ensure that every option is integer)